test/Dialect/RTL/canonicalization.mlir

```diff
@@ -498,7 +498,8 @@ rtl.module @orr_constant_folding1() -> (i1) {
 
 rtl.module @orr_constant_folding2() -> (i1) {
   %c8_i4 = comb.constant(8 : i4) : i4
-  %0 = comb.orr %c8_i4 : i4
+  %zero4 = comb.constant (0 : i4) : i4
+  %0 = comb.icmp ne %c8_i4, %zero4 : i4
   rtl.output %0 : i1
 }
 
```

Comment:
These tests are really testing `imcp ne` folding (which as far as I can tell isn't tested anywhere currently?). I'm fine to remove these (and add in explicit tests of `icmp ne` and probably a lot of other things) in a separate PR. At minimum, can you rename the tests so they're not about `orr_constant_folding`?

Reply:
Removed the orr folding tests. Will create a new PR with relevant canonicalization tests.